src/Plugin/Commerce/PaymentGateway/Heartland.php

```diff
@@ -59,13 +59,25 @@ class Heartland extends OnsitePaymentGatewayBase implements OnsiteInterface
      */
     public function __construct(array $configuration, $plugin_id, $plugin_definition, EntityTypeManagerInterface $entity_type_manager, PaymentTypeManager $payment_type_manager, PaymentMethodTypeManager $payment_method_type_manager, TimeInterface $time)
     {
-        parent::__construct($configuration, $plugin_id, $plugin_definition, $entity_type_manager, $payment_type_manager, $payment_method_type_manager, $time);
-  
+        parent::__construct($configuration, $plugin_id, $plugin_definition, $entity_type_manager, $payment_type_manager, $payment_method_type_manager, $time);       
+    }
+
+    protected function authenticate()
+    {
+        $secretKey = $this->configuration['secret_key'];
+        $env = explode('_', $secretKey)[1];
```

Comment:
What happens if the merchant enters a value for secret key that does not include an underscore? What does that look like to them?